### ISSUE TITLE
[Bug]: [ROCm] v0.26.0 release image: NaN logits from AITER fused-MoE path on gfx942 (Qwen3.5-397B-A17B-FP8) — fixed on main, requesting 0.26.x backport

### ISSUE THREAD

**Issue by dvellon** [vllm-project/vllm#51580]
### Your current environment

Environment captured from inside the serving container at test time (the
GPU node is since torn down, so this is assembled from the captured
environment record rather than a live collect_env.py run):

Image: vllm/vllm-openai-rocm@sha256:5709fafe47123becb2f5e61c32d0b97beff1a629bb40bb753c15464f69a97a18
       (:latest as of 2026-08-09)
vLLM: 0.26.0
PyTorch: 2.11.0+gitd0c8b1f (HIP 7.2.53211)
amd-aiter: 0.1.16.post3
GPU: 8x AMD Instinct MI325X (gfx942), NPS1/SPX
ROCm kernel module: ROCk 6.19.14
Host OS: Ubuntu 24.04 (DigitalOcean GPU droplet, ml-ai image)
Python (container): 3.12
Model: Qwen/Qwen3.5-397B-A17B-FP8 (sha256-verified download from HF)
Serve: TP8, --max-model-len 65536, --max-num-seqs 8

Fix-verification environment (same host, same model):
Image: vllm/vllm-openai-rocm:nightly-f8d03e77416bf90c49acbe50e233275722f02c4b
       (built 2026-08-09, digest sha256:a06f9b39ad4d...)

### 🐛 Describe the bug

With VLLM_ROCM_USE_AITER=1 on the v0.26.0 release image, serving
Qwen/Qwen3.5-397B-A17B-FP8 (TP8) on 8x MI325X (gfx942): the first
generated token is correct with a finite logprob, and every subsequent
decode step emits token id 0 — argmax over NaN logits. Prefill is
healthy; the NaN appears from the second decode step onward. Already
fixed on current main (verified below) but live in the release image.

Component isolation (separate boots, fixed-prompt exact-token-id probe):

| Configuration | Result |
|---|---|
| AITER=1, --kv-cache-dtype fp8_e4m3 | NaN from decode step 2 |
| AITER=1, kv-cache auto | NaN (KV dtype exonerated) |
| AITER=1, VLLM_ROCM_USE_AITER_MHA=0 | NaN (attention exonerated) |
| AITER=1, VLLM_ROCM_USE_AITER_MOE=0 | clean output, finite logprobs |
| AITER=0 | clean |

So the defect is in the AITER fused-MoE path for this model on gfx942.

Symptom detail:
- Greedy, max_tokens=1: correct token, finite logprob (" Paris", -0.7205)
  — prefill/first step fine.
- Greedy, max_tokens=12 on "1, 2, 3, 4,": token_ids
  [220, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0].
- Any request with logprobs fails:
  400: Out of range float values are not JSON compliant: nan
  (that error's ergonomics are covered by #48585 / #51026 — this issue
  is about the NaN itself).
- Boot log also prints: "aiter sampler does not support per-request
  generators; falling back to PyTorch-native".
- Note the model is a hybrid attention + Gated-DeltaNet MoE (per #35219),
  in case the hybrid state path is relevant.

Reproduction:

    docker run --rm --device=/dev/kfd --device=/dev/dri \
      --group-add video --group-add render \
      --ipc=host --shm-size=64g -v <weights>:/models:ro \
      -e VLLM_ROCM_USE_AITER=1 \
      --entrypoint vllm <image> \
      serve /models/Qwen3.5-397B-A17B-FP8 --tensor-parallel-size 8 \
      --max-model-len 65536 --max-num-seqs 8

    curl -s -X POST localhost:8000/v1/completions \
      -H 'Content-Type: application/json' \
      -d '{"model":"Qwen/Qwen3.5-397B-A17B-FP8","prompt":"1, 2, 3, 4,","max_tokens":12,"temperature":0.0,"return_token_ids":true}'
    # -> token_ids [220, 0, 0, 0, ...]

Fixed on main — backport request:

Verified on nightly-f8d03e77416bf90c49acbe50e233275722f02c4b (built
2026-08-09): same hardware, same model, VLLM_ROCM_USE_AITER=1, no
sub-flag exclusions — clean output (exact token ids, correct
continuation) and finite serialized logprobs. So this is fixed on
current main but live in v0.26.0, where the only workarounds are
VLLM_ROCM_USE_AITER=0 or VLLM_ROCM_USE_AITER_MOE=0 (both cost
throughput). Could the fix be identified and backported to a 0.26.x
patch release / image rebuild? Happy to re-test a candidate.

Related (checked before filing): #51039 (identical argmax-over-NaN
signature on Kimi-K3, different root cause — prior art, not a dup),
#35925 (earlier Qwen-AITER corruption, fixed by #37228, in 0.26.0),
ROCm/aiter#4451 and ROCm/aiter#4032 (same regime, different modes).

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.
